src/unity/python/turicreate/toolkits/style_transfer/style_transfer.py

```diff
@@ -284,92 +283,96 @@ def create(style_dataset, content_dataset, style_feature=None,
-                        gram_s_vgg = gram[indices]
-                        gram_p_vgg = _gram_matrix(p_vgg_output)
-
-                        style_losses.append(style_loss_mult * mse_loss(gram_s_vgg, gram_p_vgg))
-
-                    style_loss = _mx.nd.add_n(*style_losses)
-
-                    # Content Loss between content image and stylized image
-                    # Lc = L2 norm at a single layer in vgg16
-                    content_loss = _content_loss_mult * mse_loss(c_content_layer,
-                                                                 p_content_layer)
-
-                    curr_content_loss.append(content_loss)
-                    curr_style_loss.append(style_loss)
-                    # Divide loss by large number to get into a more legible
-                    # range
-                    total_loss = (content_loss + style_loss) / 10000.0
-                    Ls.append(total_loss)
-                for L in Ls:
-                    L.backward()
-
-            cur_loss = _np.mean([L.asnumpy()[0] for L in Ls])
-
-            if smoothed_loss is None:
-                smoothed_loss = cur_loss
-            else:
-                smoothed_loss = 0.9 * smoothed_loss + 0.1 * cur_loss
-            iterations += 1
-            trainer.step(batch_size)
-
-            if verbose and iterations == 1:
-                # Print progress table header
-                column_names = ['Iteration', 'Loss', 'Elapsed Time']
-                num_columns = len(column_names)
-                column_width = max(map(lambda x: len(x), column_names)) + 2
-                hr = '+' + '+'.join(['-' * column_width] * num_columns) + '+'
-                print(hr)
-                print(('| {:<{width}}' * num_columns + '|').format(*column_names, width=column_width-1))
-                print(hr)
-
-            cur_time = _time.time()
-            if verbose and (cur_time > last_time + 10 or iterations == max_iterations):
-                # Print progress table row
-                elapsed_time = cur_time - start_time
-                print("| {cur_iter:<{width}}| {loss:<{width}.3f}| {time:<{width}.1f}|".format(
-                    cur_iter = iterations, loss = smoothed_loss,
-                    time = elapsed_time , width = column_width-1))
-                if params['print_loss_breakdown']:
-                    print_content_loss = _np.mean([L.asnumpy()[0] for L in curr_content_loss])
-                    print_style_loss = _np.mean([L.asnumpy()[0] for L in curr_style_loss])
-                    print('Total Loss: {:6.3f} | Content Loss: {:6.3f} | Style Loss: {:6.3f}'.format(cur_loss, print_content_loss, print_style_loss))
-                last_time = cur_time
-            if iterations == max_iterations:
-                print(hr)
-                break
+        try:
+            c_batch = content_images_loader.next()
+        except StopIteration:
+            print('Number of content images is less than the batch_size')
+            raise
```

Comment:
So this situation is necessarily a fatal error? Why can't we handle it more gracefully? What will happen when the `StopIteration` is re-raised?

Reply:
Currently, StopIteration will be raised only when the number of content images is less than the batch_size.  
I don't see any other place else that StopIteration could be raised